HazelcastCloudTests/csharphazelcastcloudtests/CloudTests/StandardClusterTests.cs

```diff
@@ -0,0 +1,115 @@
+
+            await RcClient.LoginToHazelcastCloudUsingEnvironment();
+
+            _sslEnabledCluster = await RcClient.CreateHazelcastCloudStandardCluster(hzVersion, true);
+            _sslDisabledCluster = await RcClient.CreateHazelcastCloudStandardCluster(hzVersion, false);
+            //_sslDisabledCluster = await RcClient.GetHazelcastCloudCluster("1584");
+            //_sslEnabledCluster = await RcClient.GetHazelcastCloudCluster("1565");
+        }
+
+        [OneTimeTearDown]
+        public async Task DeleteClusters()
+        {
+            if (!string.IsNullOrEmpty(_sslEnabledCluster?.Id))
+                await RcClient.DeleteHazelcastCloudCluster(_sslEnabledCluster.Id);
+
+            if (!string.IsNullOrEmpty(_sslDisabledCluster?.Id))
+                await RcClient.DeleteHazelcastCloudCluster(_sslDisabledCluster.Id);
+        }
+
+        [TestCase(true, true)]
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        [TestCase(false, false)]
+        [Timeout(300_000)]
+        public async Task TestCloudConnection(bool isSmart, bool isSslEnabled)
+        {
+            HazelcastOptions options;
+            if (isSslEnabled)
+            {
+                options = Helper.CreateClientConfigWithSsl(_sslEnabledCluster.NameForConnect, _sslEnabledCluster.Token, isSmart,
+                    _sslEnabledCluster.CertificatePath, _sslEnabledCluster.TlsPassword);
+                _tmpClusterObject = _sslEnabledCluster;
+            }
+            else
+            {
+                options = Helper.CreateClientConfigWithoutSsl(_sslDisabledCluster.NameForConnect, _sslDisabledCluster.Token,
+                    isSmart);
+                _tmpClusterObject = _sslDisabledCluster;
+            }
+
+            var logger = options.LoggerFactory.Service.CreateLogger("test");
+
+            logger.LogInformation("Create client");
+            await using var client = await HazelcastClientFactory.StartNewClientAsync(options);
+            await using var map = await client.GetMapAsync<string, string>("MapForTest");
+            await Helper.MapPutGetAndVerify(map, logger);
+
+            logger.LogInformation("Scale up cluster from 2 node to 4");
+            await RcClient.SetHazelcastCloudClusterMemberCount(_tmpClusterObject.Id, 4);
+            await Helper.MapPutGetAndVerify(map, logger);
+
+            logger.LogInformation("Scale down cluster from 4 node to 2");
+            await RcClient.SetHazelcastCloudClusterMemberCount(_tmpClusterObject.Id, 2);
+            await Helper.MapPutGetAndVerify(map, logger);
+
+            logger.LogInformation("Stop cluster");
+            await RcClient.StopHazelcastCloudCluster(_tmpClusterObject.Id);
+
+            logger.LogInformation("Resume cluster");
+            await RcClient.ResumeHazelcastCloudCluster(_tmpClusterObject.Id);
+
+            logger.LogInformation("Wait 5 seconds to be sure client is connected again");
+            await Task.Delay(5_000);
```

Comment:
how can we be sure that 5 seconds is enough?

Reply:
We can't but it could be a sign to QA.